src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -458,10 +458,17 @@ internal static string GetSharedListenerIdOrNull(ITriggerBinding triggerBinding)
 
         internal static void CheckRetrySupport(ITriggerBinding triggerBinding, FunctionDescriptor descr, ILogger logger)
         {
-            if (descr.RetryStrategy != null && triggerBinding != null && triggerBinding.GetType().GetCustomAttribute<SupportsRetryAttribute>() == null)
+            if (descr.RetryStrategy != null)
             {
-                logger?.LogWarning($"Retries are not supported by the trigger binding for { descr.ShortName}.");
-                descr.RetryStrategy = null; // Need to fix tests
+                logger?.LogDebug($"Retry strategy is defined for function '{descr.ShortName}' with trigger binding type '{triggerBinding.GetType().Name}' + " +
+                    $"Strategy = '{JsonConvert.SerializeObject(descr.RetryStrategy)}'");
```

Comment:
We should add a guard around this to not log if we're not debug logging enabled (e.g. `if (logger?.IsEnalbed(LogLevel.Debug)` wrapper), and prefer a single formattable string (rather than 2 concatenated together). This is because we pay the price for the string formatting and JSON serialization _even if we're not logging_ as is (guard helps there), but additionally in .NET 6 we can get some advantages with overloads we'll add to `LogDebug()` that don't format the string unless it's enabled either.

Reply:
We always log to FunctionsLogs(kusto logs) whatever log level is, so the guard would be redundant